dvc/cache/base.py

```diff
@@ -89,7 +89,26 @@ def load_dir_cache(self, hash_info):
 
         return DirInfo.from_list(d)
 
-    def changed(self, path_info, hash_info):
+    def _get_filtered_hash_info(self, filter_info, hash_info, path_info):
+        dir_info = self.get_dir_cache(hash_info)
+        hash_key = filter_info.relative_to(path_info).parts
+
+        # Check whether it is a file that exists on the trie
+        if hash_key in dir_info.trie:
+            return dir_info.trie[hash_key]
+
+        depth = len(hash_key)
+        filtered_dir_info = DirInfo()
+        try:
+            for key, value in dir_info.trie.items(hash_key):
+                filtered_dir_info.trie[key[depth:]] = value
+        except KeyError:
+            return None
+        else:
+            filtered_hash_info, _ = self._get_dir_info_hash(filtered_dir_info)
+            return filtered_hash_info
```

Comment:
Just a bit flatter, since we `return` in `except` anyway.
```suggestion

        filtered_hash_info, _ = self._get_dir_info_hash(filtered_dir_info)
        return filtered_hash_info
```